### DISCUSSION TITLE
Access relationships data from the mutateFormDataBeforeCreate method

### DISCUSSION THREAD

**Discussion by gabrielchavezme** [filamentphp/filament#9837] [Ideas]
When you define a Repeater with the ->relationship('your-relationship') method you cannot access the relationship information in the mutateFormDataBeforeCreate() method.

Form definition:

```
Repeater::make('items')
    ->label('Items')
    ->columnSpanFull()
    ->cloneable()
    ->collapsible()
    ->minItems(1)
    ->reorderable(false)
    ->relationship('items')
    ->schema([

        Select::make('product_id')
            ->label('Product')
            ->required(),

        TextInput::make('qty')
            ->label('Quantity')
            ->integer()
            ->required()
            ->default(1)
    ]);
```

mutateFormDataBeforeCreate method

```
protected function mutateFormDataBeforeCreate($data) 
{
    $data['items'] // return Undefined array key "items"
}
```

If you want to access the items (relationship) you cannot because it is not included in the $data variable.

The idea of feature:

That you can access relationship data before creating them.

This would be useful for:
- Process migration data
- Validate data

For example, I want to obtain the relationship data before saving it in the database to validate the relationship information (items) with an external API.

**Comment by leandrocfe** [comment]
I think you can use `->dehydrated(true)` in this case and save the relationship manually

**Reply by Nes-cmd** [reply]
Yeah, It worked. Thanks

**Reply by OccTherapist** [reply]
Awesome, thanks for this nugget.
